src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -918,10 +922,139 @@ private void IsSelectedChanged(object sender, PropertyChangedEventArgs e)
         /// <returns> Returns a list with a maximum MaxNumSearchResults elements.</returns>
         /// <param name="search"> The search query </param>
         /// <param name="subset">Subset of nodes that should be used for the search instead of the complete set of nodes. This is a list of NodeSearchElement types</param>
-        internal IEnumerable<NodeSearchElementViewModel> Search(string search, IEnumerable<NodeSearchElement> subset = null)
+        public IEnumerable<NodeSearchElementViewModel> Search(string search, IEnumerable<NodeSearchElement> subset = null)
```

Comment:
Let's try to perserve the old API so legacy tests still work? Unless you tested locally they still pass?

Reply:
@QilongTang fixed, I re-added the old-API and modified the current API with a temporary flag, also I added the change that if is in TestMode then don't initialize LuceneNET.
Now all the tests under LibraryTest TestFixture are passing.
commit: https://github.com/DynamoDS/Dynamo/pull/14013/commits/11c11cff45f0322622ffaca565e02b4a113cede6
![image](https://github.com/DynamoDS/Dynamo/assets/61755417/5104a7d6-55c8-411c-bb51-8bf817df532c)